src/Datadog.Trace/Util/UriHelpers.cs

```diff
@@ -135,6 +172,84 @@ public static string GetCleanUriPath(string absolutePath)
 #endif
         }
 
+#if NETCOREAPP
```

Comment:
I'd like to see some benchmark numbers show that the extra complexity of having two copies of the same method are worth it.

Reply:
😆 yep, good call, turns out totally not worth it! 

```bash
BenchmarkDotNet=v0.12.1, OS=Windows 10.0.19042
Intel Core i7-8750H CPU 2.20GHz (Coffee Lake), 1 CPU, 12 logical and 6 physical cores
.NET Core SDK=5.0.201
  [Host]     : .NET Core 3.1.13 (CoreCLR 4.700.21.11102, CoreFX 4.700.21.11602), X64 RyuJIT
  DefaultJob : .NET Core 3.1.13 (CoreCLR 4.700.21.11102, CoreFX 4.700.21.11602), X64 RyuJIT


|                             Method |     Mean |   Error |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|----------------------------------- |---------:|--------:|--------:|-------:|------:|------:|----------:|
|     StringBased_LongUrlWithChanges | 320.7 ns | 4.63 ns | 4.10 ns | 0.0339 |     - |     - |     160 B |
|       SpanBased_LongUrlWithChanges | 268.1 ns | 3.70 ns | 3.63 ns | 0.0339 |     - |     - |     160 B |
|  StringBased_LongUrlWithoutChanges | 241.2 ns | 1.18 ns | 1.04 ns | 0.0353 |     - |     - |     168 B |
|    SpanBased_LongUrlWithoutChanges | 207.5 ns | 1.83 ns | 1.62 ns | 0.0355 |     - |     - |     168 B |
|    StringBased_ShortUrlWithChanges | 137.8 ns | 1.90 ns | 1.68 ns | 0.0153 |     - |     - |      72 B |
|      SpanBased_ShortUrlWithChanges | 114.7 ns | 2.19 ns | 1.83 ns | 0.0153 |     - |     - |      72 B |
| StringBased_ShortUrlWithoutChanges | 134.7 ns | 2.40 ns | 2.00 ns | 0.0169 |     - |     - |      80 B |
|   SpanBased_ShortUrlWithoutChanges | 118.9 ns | 0.83 ns | 0.65 ns | 0.0169 |     - |     - |      80 B |
```

I'll remove the span-based implementation in favour of the string-based one

Reply:
@lucaspimentel I simplified the implementation significantly in https://github.com/DataDog/dd-trace-dotnet/pull/1327/commits/1a1704d2b2ee4534465b65354525d6470130913a. As you suspected, there was no need for two copies of the same method, and in fact we could significantly simplify the "main" method too. Added some benchmarks to the description

Reply:
Awesome, thanks!